### DISCUSSION TITLE
Add more extensions

### DISCUSSION THREAD

**Discussion by sbendhiba** [Poeschl/Hassio-Addons#267] [Q&A] [answered]
Hello,

I was wondering if there is a possibility to add more extensions? Spotify for example

Thanks

**Comment by Poeschl** [comment, marked as answer]
I really would like to add the Spotify extension to Mopidy. But right now its not possible due to limitations of the library the extension is using.

**Reply by rdrr** [reply]
Hi Poeschl, thank you so so much for this.  I had thought mopidy was a no-go for my RPI4!  If I did want more extensions, like say Mopidy-SomaFM, would cloning your repository and editing the dockerfile, mirroring the how you've added the other extensions, be a good starting point?  Sorry, I'm very new.  I promise not to come back here with every error I get :), but I'd appreciate it if someone could confirm that this is a worthwhile approach.  Thanks again!

**Reply by Poeschl** [reply]
Hi, rdrr. 😉
This is a very good starting point.
For testing it out grab the folder of the addon files and copy it on the `/addon` share. Then you can install your modified addon via the supervisor pages.

**Reply by rdrr** [reply]
Thanks Poeschl!  I'll give it a shot.  Going to try for Mopidy-dLeyna, Mopidy-SomaFM, and maybe Mopdy-Jellyfin.  One at a time of course.  Thanks again, I'll report back how it goes in case it can help.

**Reply by RomanKlein777** [reply]
hi rdrr!
I am interested in results too. Especially having mopidy with spotify on home assistant would be a killing feature. Any progress?

What a feeling when I have discovered i can play spotify with spotcast, then stream radio with mpd. Having all integrated in mopidy makes it full sense - no need to stop one player and start another one....

**Reply by rdrr** [reply]
Hi @RomanKlein777, so I've run into a few issues.  I was able to get it to install, I basically cloned Peoschl's repository, copied the Mopidy folder to my /addons directory , and edited all the versions of the packages to the latest alpine package in main and community [here](https://dl-cdn.alpinelinux.org/alpine/edge/).  Then I added the extra Mopidy extensions by mirroring the other ones, changing where approriate of course.  Check out what I did [here](https://github.com/rdrr/hassio-addons/tree/master/mopidy).

I didn't add Mopidy-Spotify yet and had a few issues running.  Dlenya seemed to have crashed with a message about mapping, and SomaFM wouldn't play giving me error messages about not have the proper codecs and my gstreamer install not being correct.  I thought py3-gst solved this, but may have to add gstreamer and the gst-plugins.

I actually reinstalled Home Assistant from HASSOS on an RPI4, to Home Assistant Supervised on Raspbian OS on the same RPI4.  So, honestly, now I'm leaning toward installing mopidy and snapserver directly on raspbian, now that I can.  This was a great tutorial for me in creating my first 'custom' addon, but I'm not sure I need it in this case.

Thank you Poeschl for encouraging me and I encourage everyone else to try as well.  Poeschl has made an awesome skeleton to tinker and learn about addons. Good Luck!
